### PR TITLE
Add "support" for npcids to the slayer plugin and add dwarf guards to the dwarves task.

### LINKED ISSUE #6458: Highlight dwarven guards for slayer dwarf task
https://i.gyazo.com/143e5904eb5126b763dc3b16677a1cc6.png

something minor but, guards aren't selected while "highlighting targets" on the slayer tab, with dwarfs as the task

### PR DESCRIPTION
**Pull request by 15987632**
closes #6458
This probably wont get merged because its pretty hacky and I did it in a stupid way. The problem was that the dwarf guard has the name "Guard" and many non dwarf guards share the same name and the way the slayer plugin currently handles the slayer monsters that get highlighted for a task it wasn't possible to distinguish between the types of guards.

edit: It's not stupid anymore

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/Task.java [hunk 1]
```diff
@@ -117,55 +234,188 @@
...
+	TROLLS("Trolls", ItemID.TROLL_GUARD, NpcID.TROLL, NpcID.TROLL_2833, NpcID.TROLL_8470, NpcID.TROLL_8471,
+			NpcID.TROLL_8472, NpcID.TROLL_8473, NpcID.TROLL_GENERAL, NpcID.TROLL_GENERAL_4121, NpcID.TROLL_GENERAL_4122,
+			NpcID.THROWER_TROLL, NpcID.THROWER_TROLL_932, NpcID.THROWER_TROLL_933, NpcID.THROWER_TROLL_934,
+			NpcID.THROWER_TROLL_935, NpcID.THROWER_TROLL_4135, NpcID.THROWER_TROLL_4136, NpcID.THROWER_TROLL_4137,
+			NpcID.THROWER_TROLL_4138, NpcID.THROWER_TROLL_4139),
+	TUROTH("Turoth", ItemID.TUROTH, NpcID.TUROTH, NpcID.TUROTH_427, NpcID.TUROTH_428, NpcID.TUROTH_429,
+			NpcID.TUROTH_430, NpcID.TUROTH_431, NpcID.TUROTH_432),
+	TZHAAR("Tzhaar", ItemID.ENSOULED_TZHAAR_HEAD, NpcID.TZHAARHUR, NpcID.TZHAARHUR_2162, NpcID.TZHAARHUR_2163,
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/Task.java]: This should include fight caves/inferno wave monsters and bosses

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 2]
```diff
@@ -524,44 +525,31 @@ private boolean doubleTroubleExtraKill()
...
-		if (name == null)
-		{
-			return false;
-		}
-
-		name = name.toLowerCase();
+		int npcId = npc.getId();
 
-		for (String target : targetNames)
+		for (int targetId : targetIds)
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: return targetIds.contains

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/Task.java [hunk 3]
```diff
@@ -117,55 +234,188 @@
...
+	TROLLS("Trolls", ItemID.TROLL_GUARD, NpcID.TROLL, NpcID.TROLL_2833, NpcID.TROLL_8470, NpcID.TROLL_8471,
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/Task.java]: This should include the various "greater" trolls (such as Stick and Rock), as well as ice trolls and river trolls.

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/Task.java [hunk 4]
```diff
@@ -117,55 +234,188 @@
...
+			NpcID.SKELETON_1687, NpcID.SKELETON_1688, NpcID.SKELETON_1785, NpcID.SKELETON_2520, NpcID.SKELETON_2521,
+			NpcID.SKELETON_2522, NpcID.SKELETON_2523, NpcID.SKELETON_2524, NpcID.SKELETON_2525, NpcID.SKELETON_2526,
+			NpcID.SKELETON_3565, NpcID.SKELETON_3584, NpcID.SKELETON_3972, NpcID.SKELETON_3973, NpcID.SKELETON_3974,
+			NpcID.SKELETON_5237, NpcID.SKELETON_6441, NpcID.SKELETON_6442, NpcID.SKELETON_6443, NpcID.SKELETON_6444,
+			NpcID.SKELETON_6445, NpcID.SKELETON_6446, NpcID.SKELETON_6447, NpcID.SKELETON_6448, NpcID.SKELETON_6467,
+			NpcID.SKELETON_6468, NpcID.SKELETON_7265, NpcID.SKELETON_8070, NpcID.SKELETON_8071, NpcID.SKELETON_8072,
+			NpcID.SKELETON_8139, NpcID.SKELETON_8140, NpcID.SKELETON_BRUTE),
+	SMOKE_DEVILS("Smoke devils", ItemID.SMOKE_DEVIL, NpcID.SMOKE_DEVIL, NpcID.SMOKE_DEVIL_6639, NpcID.SMOKE_DEVIL_6655,
+			NpcID.SMOKE_DEVIL_8482, NpcID.SMOKE_DEVIL_8483, NpcID.NUCLEAR_SMOKE_DEVIL),
+	SPIDERS("Spiders", ItemID.HUGE_SPIDER, NpcID.SPIDER, NpcID.SPIDER_3019, NpcID.SPIDER_3235, NpcID.SPIDER_4561,
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/Task.java]: This should include venanatis

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/Task.java [hunk 5]
```diff
@@ -117,55 +234,188 @@
...
-	SPIRITUAL_CREATURES("Spiritual creatures", ItemID.DRAGON_BOOTS, NpcID.SPIRITUAL_MAGE, NpcID.SPIRITUAL_MAGE_2244, NpcID.SPIRITUAL_MAGE_3168, NpcID.SPIRITUAL_MAGE_3161, NpcID.SPIRITUAL_RANGER, NpcID.SPIRITUAL_RANGER_2242, NpcID.SPIRITUAL_RANGER_3160, NpcID.SPIRITUAL_RANGER_3167, NpcID.SPIRITUAL_WARRIOR, NpcID.SPIRITUAL_WARRIOR_2243, NpcID.SPIRITUAL_WARRIOR_3159, NpcID.SPIRITUAL_WARRIOR_3166),
-	STEEL_DRAGONS("Steel dragons", ItemID.STEEL_DRAGON, NpcID.STEEL_DRAGON, NpcID.STEEL_DRAGON_274, NpcID.STEEL_DRAGON_275, NpcID.STEEL_DRAGON_7255, NpcID.STEEL_DRAGON_8086),
-	SUQAHS("Suqahs", ItemID.SUQAH_TOOTH, NpcID.SUQAH, NpcID.SUQAH_788, NpcID.SUQAH_789, NpcID.SUQAH_790, NpcID.SUQAH_791, NpcID.SUQAH_792, NpcID.SUQAH_793),
+	SKELETAL_WYVERNS("Skeletal wyverns", ItemID.SKELETAL_WYVERN, NpcID.SKELETAL_WYVERN, NpcID.SKELETAL_WYVERN_466,
+			NpcID.SKELETAL_WYVERN_467, NpcID.SKELETAL_WYVERN_468),
+	SKELETONS("Skeletons", ItemID.SKELETON_GUARD, NpcID.SKELETON, NpcID.SKELETON_71, NpcID.SKELETON_72,
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/Task.java]: This should include Vet'ion

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 6]
```diff
@@ -169,7 +170,7 @@
 	private int cachedXp;
 	private Instant infoTimer;
 	private boolean loginFlag;
-	private List<String> targetNames = new ArrayList<>();
+	private Set<Integer> targetIds = new HashSet<>();
```
[the remark this answers is not in the document]
Comment (15987632) [reply runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: done

File: runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java [hunk 7]
```diff
@@ -524,44 +525,23 @@ private boolean doubleTroubleExtraKill()
 
 	private boolean isTarget(NPC npc)
 	{
-		if (targetNames.isEmpty())
+		if (targetIds.isEmpty())
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/slayer/SlayerPlugin.java]: This empty check is an unnecessary optimization now

**Comment by daviscook477** [comment]
GUARD_BANDIT should be on the bandit task

**Comment by 15987632** [comment]
> what do you mean by just support for item ids?

I meant NPC ids, just said the wrong thing.

**Comment by 15987632** [comment]
I'm going to close this for now as I haven't been working on this and it would be easier to start over anyways.

**Pull request closed without merging**